src/engine/Server.cpp

```diff
@@ -137,8 +137,8 @@ void Server::run(const string& indexBaseName, bool useText, bool usePatterns,
 
   // First set up the HTTP server, so that it binds to the socket, and
   // the "socket already in use" error appears quickly.
-  auto httpServer =
-      HttpServer{port_, "0.0.0.0", numThreads_, std::move(httpSessionHandler)};
+  auto httpServer = HttpServer{port_, "0.0.0.0", static_cast<int>(numThreads_),
```

Comment:
While you are doing small improvements, can you also change that parameter of the HttpServer to be an unsigned type, s.t. we don't need a static_cast here?

Reply:
The `HttpServer` unfortunately passes this parameter to the io context which accepts a signed int, so at some point it needs to be cast to int.

See https://www.boost.org/doc/libs/1_81_0/doc/html/boost_asio/reference/io_context/io_context.html